server/datastore/mysql/microsoft_mdm.go

```diff
@@ -138,3 +139,50 @@ WHERE
 
 	return &res, nil
 }
+
+func (ds *Datastore) GetMDMWindowsBitLockerStatus(ctx context.Context, host *fleet.Host) (*fleet.DiskEncryptionStatus, error) {
+	if host == nil {
+		return nil, errors.New("host cannot be nil")
+	}
+
+	if host.MDMInfo != nil && host.MDMInfo.IsServer {
+		return nil, nil
+	}
+
+	enabled, err := ds.getConfigEnableDiskEncryption(ctx, host.TeamID)
+	if err != nil {
+		return nil, err
+	}
+	if !enabled {
+		return nil, nil
+	}
+
+	// Note verifying, action_required, and removing_enforcement are not applicable to Windows hosts
+	stmt := fmt.Sprintf(`
+SELECT
+	CASE
+		WHEN %s THEN '%s'
+		WHEN %s THEN '%s'
+		WHEN %s THEN '%s'
+	END AS status
+FROM
+	hosts h
+	LEFT JOIN host_disk_encryption_keys hdek ON h.id = hdek.host_id
+	LEFT JOIN host_mdm hmdm ON h.id = hmdm.host_id
+WHERE
+	h.id = ? AND h.platform = 'windows'`,
```

Comment:
Just a thought, since this receives a `Host` struct, the `windows` platform check could be done above (line 148) and it wouldn't need to use the table `hosts`, I believe? The `host_mdm` could become the entry point and left join on `hdek`? Unless of course we do want to fail with "sql.ErrNoRows" if the host happens to be non-windows (which is how it behaves atm).